deployment/postgres-init-db/sql/tables/merlin/external_events/external_event.sql

```diff
@@ -0,0 +1,53 @@
+-- Create table for external events
+create table merlin.external_event (
+    key text not null,
+    event_type_name text not null,
+	  source_key text not null,
+    derivation_group_name text not null,
+    start_time timestamp with time zone not null,
+    duration interval not null,
+    properties jsonb,
+
+    constraint external_event_pkey
+      primary key (key, source_key, derivation_group_name, event_type_name),
```

Comment:
I'm not a fan of large composite primary keys, especially when one or all of the components are strings. Please make a column `id integer generated by default as identity` and set that as the primary key. I don't think it'll be a very impactful change.

Reply:
I haven't had a chance to take a look and form an opinion, but I'd like to leave as a reminder that this pk would need to be replaced with a `unique` constraint if this change is made.

Reply:
I believe Pranav et al started with synthetic keys for everything, and then based on an initial review (with @Mythicaeda ?) refactored everything to use natural keys instead.

Sounds like maybe there's a middle ground where sometimes we prefer one or the other? @JoelCourtney and @Mythicaeda , if you have some time, talk this over amongst yourselves and see if you can come to a consensus on the guidance to give Pranav

Reply:
Following a discussion with @Mythicaeda, we have decided to leave it as is, as there is no logical meaning behind a synthetic key here; using just this primary key versus a synthetic key and a unique constraint is more logically meaningful and is therefore preferred.